src/truststore/_macos.py

```diff
@@ -286,20 +293,37 @@ def _verify_peercerts_impl(
     server_hostname: str | None = None,
 ) -> None:
     certs = None
-    policy = None
+    policies = None
     trust = None
     cf_error = None
     try:
         if server_hostname is not None:
             cf_str_hostname = None
             try:
                 cf_str_hostname = _bytes_to_cf_string(server_hostname.encode("ascii"))
-                policy = Security.SecPolicyCreateSSL(True, cf_str_hostname)
+                ssl_policy = Security.SecPolicyCreateSSL(True, cf_str_hostname)
             finally:
                 if cf_str_hostname:
                     CoreFoundation.CFRelease(cf_str_hostname)
         else:
-            policy = Security.SecPolicyCreateSSL(True, None)
+            ssl_policy = Security.SecPolicyCreateSSL(True, None)
+
+        policies = ssl_policy
+        if ssl_context.verify_flags & ssl.VERIFY_CRL_CHECK_CHAIN:
+            # Add explicit policy requiring positive revocation checks
+            policies = CoreFoundation.CFArrayCreateMutable(
+                CoreFoundation.kCFAllocatorDefault,
+                0,
+                ctypes.byref(CoreFoundation.kCFTypeArrayCallBacks),
+            )
+            CoreFoundation.CFArrayAppendValue(policies, ssl_policy)
+            revocation_policy = Security.SecPolicyCreateRevocation(
```

Comment:
@davisagli Do we need to `CFRelease` the `ssl_policy` and `revocation_policy` objects below as well?

Reply:
@sethmlarson Yes, having reviewed docs about CoreFoundation memory management more closely, I think you're right.